From 2853977b8095486b7fff3424d473aba45a3df211 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:39:36 -0400
Subject: [ethernet] i40e: Avoid adding the TCP-IPv4 filter twice

Message-id: <1409658042-22689-216-git-send-email-sassmann@redhat.com>
Patchwork-id: 90401
O-Subject: [RHEL7.1 PATCH 215/281] i40e: Avoid adding the TCP-IPv4 filter twice
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Anjali Singhai Jain <anjali.singhai@intel.com>

There wasn't a need to play the logic twice, it seems
like a left over from when we had to add two PTYPEs for
one filter. There should be no change in the number of
filters that actually got added to the hardware.

Change-ID: I5071d02eafd020b60e30eb96219f110f334eec85
Signed-off-by: Anjali Singhai Jain <anjali.singhai@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 12be846ddda83b9f641be6cdbfd6891594fda4ec)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_txrx.c b/drivers/net/ethernet/intel/i40e/i40e_txrx.c
index e49f31d..4d96b74 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_txrx.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_txrx.c
@@ -271,19 +271,6 @@ static int i40e_add_del_fdir_tcpv4(struct i40e_vsi *vsi,
     fd_data->pctype, ret);
  }
 
- fd_data->pctype = I40E_FILTER_PCTYPE_NONF_IPV4_TCP;
-
- ret = i40e_program_fdir_filter(fd_data, raw_packet, pf, add);
- if (ret) {
-  dev_info(&pf->pdev->dev,
-    "Filter command send failed for PCTYPE %d (ret = %d)\n",
-    fd_data->pctype, ret);
-  err = true;
- } else {
-  dev_info(&pf->pdev->dev, "Filter OK for PCTYPE %d (ret = %d)\n",
-     fd_data->pctype, ret);
- }
-
  return err ? -EOPNOTSUPP : 0;
 }
 
-- 
1.7.1